force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,193 @@
+        //List<Opportunity> opportunitiesToUpdateList = new List<Opportunity>();
+        
+        Map<Opportunity, Id> opportunitiesWithoutPrimaryContactToAccountIdMap = new Map<Opportunity, Id>();
+        for(Opportunity opp : opportunityListFromTrigger){        
+            if(opp.Primary_Contact__c == null){
+                opportunitiesWithoutPrimaryContactToAccountIdMap.put(opp, opp.AccountId);
+            }
+        }
+
+        List<Contact> vpSalesContactList = [SELECT Id, Name, Title, AccountId FROM CONTACT WHERE Title = 'VP Sales' AND AccountId IN :opportunitiesWithoutPrimaryContactToAccountIdMap.values()];
+
+        Map<Id, Id> accIdConIdMap = new Map<Id, Id>();
+
+        for(Contact con : vpSalesContactList){
+            accIdConIdMap.put(con.AccountId, con.Id);
+        }
+
+        for(Opportunity opp : opportunitiesWithoutPrimaryContactToAccountIdMap.keySet()){
+            opp.Primary_Contact__c = accIdConIdMap.get(opp.AccountId);
+            opportunitiesToUpdate.add(opp);
+            //System.debug(opportunitiesToUpdate);
+           
+        }
+        List<Opportunity> bRecords = [SELECT Id, Primary_Contact__c FROM Opportunity WHERE AccountId IN :opportunitiesWithoutPrimaryContactToAccountIdMap.values() LIMIT 5];    
+        try{
+            update opportunitiesToUpdate;
+        }catch (DmlException e) {
+            // Output the error message to the debug log
+            System.debug('An error occurred: ' + e.getMessage());
+            // Optionally, you can output more details such as:
+            System.debug('Error status code: ' + e.getDmlStatusCode(0));
+            System.debug('Error field: ' + e.getDmlFields(0));
+            
+            System.debug('Error stack trace: ' + e.getStackTraceString());
+        }
+    
+        update opportunitiesToUpdate;
+        List<Opportunity> aRecords = [SELECT Id, Primary_Contact__c FROM Opportunity WHERE AccountId IN :opportunitiesWithoutPrimaryContactToAccountIdMap.values() LIMIT 5];
+    }
+
+    public void setDefaultOpportunityType(){
+        // Set default Type for new Opportunities
+        for(Opportunity opp : triggerList){
+            if (opp.Type == null){
+                opp.Type = 'New Customer';
+            }
+        }
+    }
+
+    /*
+    * deleteCloseWonOpportunity()
+    * When an opportunity is deleted prevent the deletion of a closed won opportunity if the account industry is 'Banking'.
+    * Trigger should only fire on delete.
+    */
+    public void deleteCloseWonOpportunity(){
+        List<Id> accIds = new List<Id>();
+        for(Opportunity opp : triggerList){
+            accIds.add(opp.AccountId);
+        }
+        List<Account> relatedAccounts = [SELECT Id, Industry FROM Account WHERE Id IN :accIds];
+
+        Map<Id, String> accIdIndustryMap = new Map<Id, String>();
+
+        for(Account acc : relatedAccounts){
+            accIdIndustryMap.put(acc.Id, acc.Industry);
+        }
```

Comment:
You can query into a Map and then get the Industry from the account Map. No need to create a separate map to the string.